### PR TITLE
Resize regardless of number of dims, considering the last two dims as image

### PR DESCRIPTION
**Pull request by justanhduc**
### Description
Hi. I propose to change the behavior of `kornia.geometry.transform.resize` to support any dimensional images as discussed [here](https://kornia.slack.com/archives/CV71WGN8H/p1622448371002400). The last two dims will be considered as image height and width. I wrote a test to serve as an example. If this is approved I will work on the rest.

### Status
**Work in progress**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [x] Did you discuss the functionality or any breaking changes before ?
- [x] **Pass all tests**: did you test in local ? `make test`
- [x] Unittests: did you add tests for your new functionality ?
- [x] Documentations: did you build documentation ? `make build-docs`
- [x] Implementation: is your code well commented and follow conventions ? `make lint`
- [x] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>

  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?

</details>

### CODE REVIEW

**Review by edgarriba — APPROVED** [review]
LGTM. This is cool - I'm thinking how this could be extended to a more generic thing across the library to be more flexible in terms of supporting different tensor shapes.

**Comment by justanhduc** [comment]
Yeah true. So far as we focus mainly on images and videos, I think many of the functions can be made to work like this. I will finalize this PR and let you know once I finish.

**Comment by justanhduc** [comment]
@edgarriba Hey I think this can be wrapped up now. I'm eager to discuss if we can expand this behavior to other functions and modules.

File: test/geometry/transform/test_affine.py [hunk 1]
```diff
@@ -14,46 +14,161 @@ def test_smoke(self, device, dtype):
...
+        out = kornia.resize(inp, (3, 4))
+        assert_allclose(inp, out, atol=1e-4, rtol=1e-4)
 
     def test_upsize(self, device, dtype):
         inp = torch.rand(1, 3, 3, 4, device=device, dtype=dtype)
         out = kornia.resize(inp, (6, 8))
         assert out.shape == (1, 3, 6, 8)
 
+        # 2D
+        inp = torch.rand(3, 4, device=device, dtype=dtype)
```
Comment (edgarriba) [inline test/geometry/transform/test_affine.py:38]: fro next PR - different shape should be addressed via `pytest.mark.parametrize`

**Review by edgarriba — APPROVED** [review]

File: test/geometry/transform/test_affine.py (hunk 1 above)
Comment (justanhduc) [reply test/geometry/transform/test_affine.py:38]: Got it!

**Pull request merged**